specification/cpim/resource-manager/Microsoft.AzureActiveDirectory/preview/2020-05-01-preview/cpim.json

```diff
@@ -0,0 +1,366 @@
+        "tags": [
+          "GuestUsages"
+        ],
+        "x-ms-examples": {
+          "GuestUsages_Get": {
+            "$ref": "./examples/GuestUsagesGet.json"
+          }
+        },
+        "summary": "Gets a Guest Usages resource",
+        "description": "Gets a Guest Usages resource for the Microsoft.AzureActiveDirectory resource provider",
+        "parameters": [
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "#/parameters/ResourceNameParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK",
+            "schema": {
+              "$ref": "#/definitions/GuestUsagesResource"
+            }
+          }
+        }
+      }
+    }
+  },
+  "definitions": {
+    "Operation": {
+      "description": "Microsoft.AzureActiveDirectory REST API operation.",
+      "type": "object",
+      "properties": {
+        "name": {
+          "description": "Operation name: {provider}/{resource}/{operation}.",
+          "type": "string",
+          "readOnly": true
+        },
+        "display": {
+          "description": "The object that represents the operation.",
+          "properties": {
+            "provider": {
+              "description": "Service provider: Microsoft.AzureActiveDirectory.",
+              "type": "string",
+              "readOnly": true
+            },
+            "resource": {
+              "description": "Resource on which the operation is performed: GuestUsages, etc.",
+              "type": "string",
+              "readOnly": true
+            },
+            "operation": {
+              "description": "Operation type: Read, write, delete, etc.",
+              "type": "string",
+              "readOnly": true
+            },
+            "description": {
+              "description": "Friendly name of the operation",
+              "type": "string"
+            }
+          }
+        }
+      }
+    },
+    "OperationListResult": {
+      "description": "Result of listing operations for the resourceProvider",
+      "properties": {
+        "value": {
+          "description": "List of cpim service operations supported by the Microsoft.AzureActiveDirectory resource provider.",
+          "type": "array",
+          "readOnly": true,
+          "items": {
+            "$ref": "#/definitions/Operation"
+          }
+        }
+      }
+    },
+    "GuestUsagesResourceProperties": {
+      "description": "Guest Usages Resource Properties",
+      "type": "object",
+      "properties": {
+        "tenantId": {
+          "type": "string",
+          "description": "An identifier that represents the Guest Usages Resource Properties."
+        }
+      }
+    },
+    "GuestUsagesResource": {
```

Comment:
can the guesUsage resource belong a tenant different from which the API call has been made? tenant Id is present in the header, but its for the calling tenant;

Reply:
In future it might be possible to do this as the user could be an admin for multiple tenants and he would want to link the tenant to subscription from a single AAD tenant. Though this is not happening at the moment. THe only advantage of this happening rightnow is that it avoid us an extra call to get the tenant's tenantGuid.

Reply:
But this comes in as part of the header with every call. can you add it later when its needed?

Reply:
Details here - https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/common-api-details.md

Reply:
@animeshseth Please indicate how this comment was resolved.

Reply:
This was done in an offline conversation. The same has been mentioned by Gaurav. The crux being, the tenantId parameter would be used to create the guest usages resource for Tenant B when the user is in the context of Tenant A. All our existing API and newer apis take this as a parameter in the input and also our authorization layers uses this id to avoid making an extra call to fetch the tenant details